File: LiveDataDemo/app/src/main/java/guandong/com/livedatademo/repository/MyRepositoryImpl.java
package guandong.com.livedatademo.repository;

import android.arch.lifecycle.MutableLiveData;

import java.util.ArrayList;
import java.util.List;

/**
 * @author guandongchen
 * @date 26/03/2018
 */

public class MyRepositoryImpl implements MyRepository{
    @Override
    public MutableLiveData<List<String>> getLiveData(){
        MutableLiveData<List<String>> listAllMutableLiveData = new MutableLiveData<>();
        List<String> tempList =new ArrayList<>();
        tempList.add("1");
        tempList.add("2");
        listAllMutableLiveData.setValue(tempList);
        return listAllMutableLiveData;
    }
    @Override
    public List<String> getAllData() {
        return getLiveData().getValue();
    }

}

File: LiveDataDemo/app/src/main/java/guandong/com/livedatademo/repository/MyRepositoryImpl.kt
package guandong.com.livedatademo.repository

import android.arch.lifecycle.MutableLiveData

import java.util.ArrayList

/**
 * @author guandongchen
 * @date 26/03/2018
 */

class MyRepositoryImpl : MyRepository {
    override val liveData: MutableLiveData<List<String>>
        get() {
            val listAllMutableLiveData = MutableLiveData<List<String>>()
            val tempList = ArrayList<String>()
            tempList.add("1")
            tempList.add("2")
            listAllMutableLiveData.value = tempList
            return listAllMutableLiveData
        }
    override val allData: List<String>?
        get() = liveData.value

}
